Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2198 @@
+                {
+                    //grab the delta value we care about
+                    absAxisHandDelta = (scrollDirection == ScrollDirectionType.UpAndDown) ? Mathf.Abs(handDelta.y) : Mathf.Abs(handDelta.x);
+
+                    //Catch an intentional finger in scroller to stop momentum, this isn't a drag its definitely a stop
+                    if (absAxisHandDelta > (handDeltaMagThreshold * 0.1f) || TimeTest(initialPressTime, Time.time, dragTimeThreshold))
+                    {
+                        scrollVelocity = 0.0f;
+                        avgVelocity = 0.0f;
+
+                        isDragging = true;
+                        velocityState = VelocityState.None;
+
+                        //reset initialHandPos to prevent the scroller from jumping
+                        initialScrollerPos = scrollContainer.transform.localPosition;
+                        initialHandPos = currentPointerPos;
+                    }
+                }
+
+                //Make sure we're actually (near) touched and not a pointer event, do a dot product check            
+                bool scrollRelease = UseNearScrollBoundary ? DetectScrollRelease(transform.forward * -1.0f, thresholdPoint, currentPointerPos, clippingObject.transform, transform.worldToLocalMatrix, scrollDirection)
+                                                           : DetectScrollRelease(transform.forward * -1.0f, thresholdPoint, currentPointerPos, null, null, null);
+
+                if (isTouched && scrollRelease)
+                {
+                    //We're on the other side of the original touch position. This is a release.
+                    if (!isDragging)
+                    {
+                        //Its a click release
+                        Collider[] c = focusedObject.GetComponentsInChildren<Collider>();
+                        bool isColliderActive = false;
+
+                        foreach (Collider col in c)
+                        {
+                            if (col.enabled)
+                            {
+                                isColliderActive = true;
+                                break;
+                            }
+                        }
+
+                        if (isColliderActive)
+                        {
+                            //Fire the UnityEvent
+                            ClickEvent?.Invoke(focusedObject);
+                        }
+                    }
+                    else
+                    {
+                        //Its a drag release
+                        initialScrollerPos = workingScrollerPos;
+                        velocityState = VelocityState.Calculating;
+                    }
+
+                    //Release the pointer
+                    if (currentPointer.GetType() != typeof(PokePointer))
+                    {
+                        currentPointer.IsFocusLocked = false;
+                        currentPointer = null;
```

Comment:
Also, this is only running if isTouched? So really doesn't seem necessary.

Reply:
Agree. Now that this sits on top of the MRTK Input system, I dont think we need this any more.